iroh-gateway/src/client.rs

```diff
@@ -98,8 +108,15 @@ impl<T: ContentLoader + std::marker::Unpin> Client<T> {
                 .pretty(self.resolver.clone(), OutMetrics { start: start_time })
                 .map_err(|e| e.to_string())?;
 
-            let stream = ReaderStream::new(reader);
-            let body = PrettyStreamBody(stream, metadata.size);
+            let mut buf_reader = tokio::io::BufReader::with_capacity(1024 * 1024, reader);
+            let body_sample = buf_reader
+                .fill_buf()
+                .await
+                .map_err(|e| e.to_string())?
+                .to_vec();
```

Comment:
Why not avoid the storing in a `Vec`, by doing the mime check  here, and store the result of it

Reply:
and here I am shuffling 1MB around for no reason 👍